lua/entities/gmod_wire_pod.lua

```diff
@@ -1,649 +1,692 @@
+    self:ColorByLinkStatus(b and self.LINK_STATUS_ACTIVE or self.LINK_STATUS_LINKED)
+    self.Activated = b
+    WireLib.TriggerOutput(self, "Active", b and 1 or 0)
 end
 
-function ENT:SetActivated( b )
-	if (self.Activated == b) then return end
+function ENT:HidePlayer(b)
+    if not self:HasPly() then return end
+    local c = self:GetPly():GetColor()
 
-	self:ColorByLinkStatus(b and self.LINK_STATUS_ACTIVE or self.LINK_STATUS_LINKED)
+    if b then
+        self.OldPlyAlpha = c.a
+        c.a = 0
+    else
+        c.a = self.OldPlyAlpha or 255
+        self.OldPlyAlpha = nil
+    end
 
-	self.Activated = b
-	WireLib.TriggerOutput(self, "Active", b and 1 or 0)
+    self:GetPly():SetColor(c)
+    self:GetPly():SetRenderMode(c.a ~= 255 and RENDERMODE_TRANSALPHA or RENDERMODE_NORMAL)
 end
 
-function ENT:HidePlayer( b )
-	if not self:HasPly() then return end
-
-	local c = self:GetPly():GetColor()
-	if b then
-		self.OldPlyAlpha = c.a
-		c.a = 0
-	else
-		c.a = self.OldPlyAlpha or 255
-		self.OldPlyAlpha = nil
-	end
-	self:GetPly():SetColor(c)
-	self:GetPly():SetRenderMode(c.a ~= 255 and RENDERMODE_TRANSALPHA or RENDERMODE_NORMAL)
-end
-
-function ENT:SetHidePlayer( b )
-	if (self.HidePlayer == b) then return end
-
-	self.HidePlayerVal = b
+function ENT:SetHidePlayer(b)
+    if (self.HidePlayer == b) then return end
+    self.HidePlayerVal = b
 
-	if (self:HasPly()) then
-		self:HidePlayer( b )
-	end
+    if (self:HasPly()) then
+        self:HidePlayer(b)
+    end
 end
 
-function ENT:LinkEnt( pod )
-	pod = WireLib.GetClosestRealVehicle(pod,self:GetPos(),self:GetPlayer())
+function ENT:LinkEnt(pod)
+    pod = WireLib.GetClosestRealVehicle(pod, self:GetPos(), self:GetPlayer())
+    -- if pod is still not a vehicle even after all of the above, then error out
+    if not IsValid(pod) or not pod:IsVehicle() then return false, "Must link to a vehicle" end
+    if hook.Run("CanTool", self:GetPlayer(), WireLib.dummytrace(pod), "wire_pod") == false then return false, "You do not have permission to access this vehicle" end
+    self:SetPod(pod)
 
-	-- if pod is still not a vehicle even after all of the above, then error out
-	if not IsValid(pod) or not pod:IsVehicle() then return false, "Must link to a vehicle" end
-	if hook.Run( "CanTool", self:GetPlayer(), WireLib.dummytrace(pod), "wire_pod" ) == false then return false, "You do not have permission to access this vehicle" end
+    WireLib.SendMarks(self, {pod})
 
-	self:SetPod( pod )
-	WireLib.SendMarks(self, {pod})
-	return true
+    return true
 end
+
 function ENT:UnlinkEnt()
-	if IsValid(self.Pod) then
-		self.Pod:RemoveCallOnRemove("wire_pod_remove")
-	end
-	self:SetShowCursor( 0 )
-	self.Pod = nil
-	self:PlayerExited()
-	WireLib.SendMarks(self, {})
-	WireLib.TriggerOutput( self, "Entity", NULL )
-	self:ColorByLinkStatus(self.LINK_STATUS_UNLINKED)
-	return true
+    if IsValid(self.Pod) then
+        self.Pod:RemoveCallOnRemove("wire_pod_remove")
+    end
+	table.RemoveByValue(pods, self)
```

Comment:
ok if you remove this here, and then relink the ent, wouldn't that be a problem?.

Reply:
They're spaces instead of tabs. Needs to be reverted.

Reply:
also it's far more efficient to use a lookup table instead of removebyvalue

Reply:
Well, it's more efficient to use ipairs for the loops though.

Reply:
> ok if you remove this here, and then relink the ent, wouldn't that be a problem?
> 
> The solution seems to be to move this to OnRemove.

I thought that function was only called by OnRemove, and I saw that all of the code was there so I was trying to be consistent. I moved that one function call to OnRemove instead like you suggested. :)